src/types.ts

```diff
@@ -112,8 +112,33 @@ export type EventType = "accountsChanged" | "networkChanged";
 
 export type EventHandler = (data: any) => void;
 
+// EIP-747:
+// https://github.com/ethereum/EIPs/blob/master/EIPS/eip-747.md
+interface WatchAssetParameters {
```

Comment:
EIP-747 suggests an interface agnostic typing (see [here](https://eips.ethereum.org/EIPS/eip-747#wallet_watchasset)) -
```
export type WatchAssetParameters = {
    type: string; // The asset's interface, e.g. 'ERC20'
    options: {
        address: string; // The hexadecimal Ethereum address of the token contract
        symbol?: string; // A ticker symbol or shorthand, up to 5 alphanumerical characters
        decimals?: number; // The number of asset decimals
        image?: string; // A string url of the token logo
    };
};
```
I think we should stick to that